genrl/deep/agents/ppo1/ppo1.py

```diff
@@ -132,151 +137,152 @@ def create_model(self):
         self.policy_old.load_state_dict(self.policy_new.state_dict())
 
         self.optimizer_policy = opt.Adam(
-            self.policy_new.parameters(), lr=self.lr_policy
+            list(self.policy_new.parameters())+list(self.value_fn.parameters()), lr=self.lr_policy
         )
-        self.optimizer_value = opt.Adam(self.value_fn.parameters(), lr=self.lr_value)
+        # self.optimizer_value = opt.Adam(self.value_fn.parameters(), lr=self.lr_value)
+
+        # self.traj_reward = []
+        # self.policy_old.policy_hist = Variable(torch.Tensor())
+        # self.policy_new.policy_hist = Variable(torch.Tensor())
+        # self.value_fn.value_hist = Variable(torch.Tensor())
 
-        self.traj_reward = []
-        self.policy_old.policy_hist = Variable(torch.Tensor())
-        self.policy_new.policy_hist = Variable(torch.Tensor())
-        self.value_fn.value_hist = Variable(torch.Tensor())
+        # self.policy_new.loss_hist = Variable(torch.Tensor())
+        # self.value_fn.loss_hist = Variable(torch.Tensor())
 
-        self.policy_new.loss_hist = Variable(torch.Tensor())
-        self.value_fn.loss_hist = Variable(torch.Tensor())
+        self.rollout = RolloutBuffer(2048, self.env.observation_space, self.env.action_space, n_envs=self.env.n_envs)
 
     def select_action(self, state):
         state = torch.as_tensor(state).float().to(self.device)
 
         # create distribution based on policy_old output
-        action, c_old = self.policy_old.get_action(Variable(state), deterministic=False)
-        _, c_new = self.policy_new.get_action(Variable(state), deterministic=False)
-        val = self.value_fn.get_value(Variable(state))
-
-        # store policy probs and value function for current traj
-        self.policy_old.policy_hist = torch.cat(
-            [
-                self.policy_old.policy_hist,
-                c_old.log_prob(action).exp().prod().unsqueeze(0),
-            ]
-        )
+        action, c_new = self.policy_new.get_action(state, deterministic=False)
+        _, c_old = self.policy_old.get_action(state, deterministic=False)
+        val = self.value_fn.get_value(state)
 
-        self.policy_new.policy_hist = torch.cat(
-            [
-                self.policy_new.policy_hist,
-                c_new.log_prob(action).exp().prod().unsqueeze(0),
-            ]
-        )
-
-        self.value_fn.value_hist = torch.cat(
-            [self.value_fn.value_hist, val.unsqueeze(0)]
-        )
+        return action.detach().cpu().numpy(), val, c_new.log_prob(action)
 
-        return action
+    def evaluate_actions(self, obs, old_actions):
+        val = self.value_fn.get_value(obs)
+        print(val.shape)
+        _, dist = self.policy_old.get_action(obs)
```

Comment:
This should've been: `_, dist = self.policy_new.get_action(obs)`.

@Het-Shah 

cc: @ajaysub110 (Kill me already)

Reply:
Lol. How did you even find this.

Reply:
Fell down to running python debugger on it.

Reply:
lol nice